### DISCUSSION TITLE
How to interrupt a company if a bracket is encountered?

### DISCUSSION THREAD

**Discussion by Artawower** [company-mode/company-mode#1239] [Q&A] [answered]
I use company-tabnine and company-yasnipet as backend, unlike company-capf they can suggest completion after brackets, and when i want to move carriage at a new line i automatically complete first suggestion. Its very annoying, and i don't know how to solve this problem, i want to disable popup when such symbols is encountered
<img width="335" alt="image" src="https://user-images.githubusercontent.com/16963833/137111100-49baadd6-113f-4b90-8b07-96a0b09a051c.png">

**Comment by dgutov** [comment, marked as answer]
What is your value of `company-minimum-prefix-length`?

If it's more than 0, and `(company-tabnine 'prefix)` returns `""` in the given context, you shouldn't get any completions yet.

You will also want to set `company-tabnine-always-trigger` to nil.

If the resulting behavior looks too conservative for you, you can instead set `company-idle-delay` to a function which checks the text around point for some conditions (e.g. `{` is before point AND `}` is after point), and returns the delay number *if* that's not the case.

**Reply by Artawower** [reply]
Thanks. `company-idle-delay` as function is what i've been looking for!
